transform/log_address_index.go

```diff
@@ -127,21 +119,21 @@ func (i *LogAddressIndex) matchingBlocks(addrs []eth.Address, eventSigs []eth.Ha
 		}
 		sigsBitmap.Or(i.eventSigs[sigString])
 	}
-	if addrBitmap.IsEmpty() {
+
+	if len(addrs) == 0 { // ONLY try to match signatures
 		out := sigsBitmap.ToArray()
 		if len(out) == 0 {
 			return nil
 		}
 		return out
 	}
 
-	addrBitmap.And(sigsBitmap) // transforms addrBitmap
+	addrBitmap.And(sigsBitmap) // addrBitmap is changed in-place, becomes merged data
```

Comment:
L'implémentation de la notion de "or" et "and" entre les filtres de addresses et events, a pas lieu de vive down low dans l'implémentation de l'index. Il devrait être plus haut. Ça devient visible quand on réfléchi à l'index comme un élément général, qui aurait des key/values plutôt que des "filterAddresses" et "filterEventLogs".